fenix/app/src/androidTest/java/org/mozilla/fenix/onboarding/view/JunoOnboardingMapperTest.kt

```diff
@@ -19,28 +28,200 @@ class JunoOnboardingMapperTest {
     fun showNotificationTrue_showAddWidgetFalse_pagesToDisplay_returnsSortedListOfAllConvertedPages_withoutAddWidgetPage() {
-        val expected = listOf(defaultBrowserPageUiData, syncPageUiData, notificationPageUiData)
-        assertEquals(expected, unsortedAllKnownCardData.toPageUiData(true, false))
+        listOf(defaultBrowserPageUiData, syncPageUiData, notificationPageUiData).also {
+            assertEquals(
+                it,
+                unsortedAllKnownCardData.toPageUiData(
+                    context = context,
+                    showNotificationPage = true,
+                    showAddWidgetPage = false,
+                ),
+            )
+        }
     }
 
     @Test
     fun showNotificationFalse_showAddWidgetFalse_pagesToDisplay_returnsSortedListOfConvertedPages_withoutNotificationPage_and_addWidgetPage() {
-        val expected = listOf(defaultBrowserPageUiData, syncPageUiData)
-        assertEquals(expected, unsortedAllKnownCardData.toPageUiData(false, false))
+        listOf(defaultBrowserPageUiData, syncPageUiData).also {
+            assertEquals(
+                it,
+                unsortedAllKnownCardData.toPageUiData(
+                    context = context,
+                    showNotificationPage = false,
+                    showAddWidgetPage = false,
+                ),
+            )
+        }
     }
 
     @Test
     fun showNotificationFalse_showAddWidgetTrue_pagesToDisplay_returnsSortedListOfAllConvertedPages_withoutNotificationPage() {
-        val expected = listOf(defaultBrowserPageUiData, addSearchWidgetPageUiData, syncPageUiData)
-        assertEquals(expected, unsortedAllKnownCardData.toPageUiData(false, true))
+        listOf(defaultBrowserPageUiData, addSearchWidgetPageUiData, syncPageUiData).also {
+            assertEquals(
+                it,
+                unsortedAllKnownCardData.toPageUiData(
+                    context = context,
+                    showNotificationPage = false,
+                    showAddWidgetPage = true,
+                ),
+            )
+        }
     }
 
     @Test
     fun showNotificationTrue_and_showAddWidgetTrue_pagesToDisplay_returnsSortedListOfConvertedPages() {
-        val expected = listOf(defaultBrowserPageUiData, addSearchWidgetPageUiData, syncPageUiData, notificationPageUiData)
-        assertEquals(expected, unsortedAllKnownCardData.toPageUiData(true, true))
+        listOf(
+            defaultBrowserPageUiData,
+            addSearchWidgetPageUiData,
+            syncPageUiData,
+            notificationPageUiData,
+        ).also {
+            assertEquals(
+                it,
+                unsortedAllKnownCardData.toPageUiData(
+                    context = context,
+                    showNotificationPage = true,
+                    showAddWidgetPage = true,
+                ),
+            )
+        }
+    }
+
+    @Test
+    fun cardConditionsNotEmpty_match_prerequisites_and_disqualifiers_returnsShouldDisplayCardTrue() {
```

Comment:
For these new test names could we try to match the format of GIVEN_WHEN_THEN e.g. givenSomeCondition_whenThisHappens_thenThisHappens? So tests below would look something like
```suggestion
    fun cardConditionsNotEmptyMatchPrerequisitesAndDisqualifiers_shouldDisplayCard_returnsTrue() {
```

Reply:
could maybe simplify to `noEmptyConditionsMatch...`